### PR TITLE
[179] Add suite merging functionality

### LINKED ISSUE #179: Admins can merge suites
Use case: grouping many standalone singles into a suite, adding a standalone single to an existing suite as a bedroom.

### PR DESCRIPTION
**Pull request by orenyk**
Resolves #179
- Add SuiteMergerForm form / service object
- Add OtherSuitesInBuildingQuery query object
- Add routes / view / policies for suite merger
- Update suite size counter cache callbacks in Room to handle switching the suite

### CODE REVIEW

File: spec/forms/suite_merger_form_spec.rb [hunk 1]
```diff
@@ -0,0 +1,135 @@
+# frozen_string_literal: true
+require 'rails_helper'
+
+RSpec.describe SuiteMergerForm, type: :model do
+  describe '.submit' do
+    xit 'calls #submit on a new instance of SuiteMergerForm'
+  end
+
+  describe 'validations' do
+    subject { described_class.new }
```
Comment (orenyk) [inline spec/forms/suite_merger_form_spec.rb]: delete this

File: app/forms/suite_merger_form.rb [hunk 2]
```diff
@@ -0,0 +1,135 @@
+# frozen_string_literal: true
+#
+# Form / service object for merging two suites together. Ensures suites belong
+# to the same building.
+class SuiteMergerForm
+  include ActiveModel::Model
+
+  attr_reader :other_suite_id
+  attr_writer :number
+  attr_accessor :suite, :other_suite
```
Comment (esoterik) [inline app/forms/suite_merger_form.rb]: do these need to be public?

File: app/views/suites/show.html.erb [hunk 3]
```diff
@@ -6,5 +6,11 @@
   <% end %>
 </ul>
 <p><%= link_to 'Add room', new_room_path %></p>
+<% if policy(@suite).merge? %>
```
Comment (esoterik) [inline app/views/suites/show.html.erb]: can we add some sort of header / instructions to this?

**Comment by orenyk** [comment]
Ok, this should be fixed to take a field and hopefully exhibit somewhat sane behavior when it comes to input issues. Currently the form on the suite show page will take you to the full form _even if the suite number entered is invalid_. This seemed acceptable to me since the final merge will fail if it isn't found with a reasonable error message. What do you think?

**Comment by orenyk** [comment]
Updated!

File: app/views/suites/merge.html.erb [hunk 4]
```diff
@@ -0,0 +1,12 @@
+<h1>Merge suite <%= @suite.number %></h1>
```
Comment (esoterik) [inline app/views/suites/merge.html.erb]: Can the header have both suite numbers in it?

**Pull request merged**